File: yh_tutorial/yh_tutorial_6/src/yh_client_6.cpp
#include "ros/ros.h"
#include "yh_tutorial_6/yh_srv_6.h"
#include <cstdlib>

int main(int argc, char ** argv)
{
    ros::init(argc, argv, "yh_client_6");

    if (argc != 4)
    {
        ROS_INFO("wrong command");
        return 1;
    }

    ros::NodeHandle nh;

    ros::ServiceClient service_client = nh.serviceClient<yh_tutorial_6::yh_srv_6>("yh_service_6");

    yh_tutorial_6::yh_srv_6 srv;

    srv.request.a = atoll(argv[1]);
    srv.request.b = atoll(argv[2]);
    srv.request.c = atoll(argv[3]);

    if (service_client.call(srv))
    {
        ROS_INFO("%ld", srv.response.result);
    }
    else
    {
        ROS_ERROR("Failed");
        return 1;
    }

    return 0;
}

File: yh_tutorial/yh_tutorial_6/scripts/yh_client_6.py
#!/usr/bin/python
# -*- coding: utf8 -*-

import rospy
from yh_tutorial_6.srv import yh_srv_6
import sys

def yh_client(a, b, c):
    rospy.wait_for_service("yh_service_6")
    try:
        service_client = rospy.ServiceProxy("yh_service_6", yh_srv_6)
        res = service_client(a, b, c)
        return res.result
    except rospy.ServiceException as e:
        rospy.logerr(e)

if __name__ == "__main__":
    rospy.init_node("yh_client_6")
    if len(sys.argv) != 4:
        rospy.loginfo("wrong")
        sys.exit(1)
    
    a = int(sys.argv[1])
    b = int(sys.argv[2])
    c = int(sys.argv[3])
    res = yh_client(a, b, c)
    rospy.loginfo("%d", res)